### DISCUSSION TITLE
Access param in sitemap

### DISCUSSION THREAD

**Discussion by muezz** [vercel/next.js#68171] [Help] [answered]
How can I access the dynamic param in the url in the `sitemap` or `generateSitemaps` functions?

Lets say I have `sitemap.ts` in the directory `/app/home/styles/[styleId]/`. How can I access the `styleId` param inside the `sitemap` function?

**Comment by icyJoseph** [comment]
Refer to, https://nextjs.org/docs/app/api-reference/file-conventions/metadata/sitemap#generating-multiple-sitemaps:

So, in short, within the `/app/home/styles/[styleId]/sitemap.ts` file, you need an additional function:

```ts
export async function generateSitemaps() {
  // Fetch the total number of products and calculate the number of sitemaps needed
  // make any async calls you need, etc
  return [{ styleId: 0 }, { styleId: 1 }, { styleId: 2 }, { styleId: 3 }]
}
```

**Reply by muezz** [reply]
I do know about this function but this implies that I have to "fetch" all styleIds from the DB. I feel like I should be able to access the styleId from the url and generate only the urls that are nested inside that particular styleId.

You method means that I will have to generate ALL nested URLs in the same sitemap function which is not a bad thing but I would prefer to split it.

**Reply by icyJoseph** [reply, marked as answer]
I understand what you want, but that's still not possible, IIRC. Hence not present in the docs.

You can still make a `app/my/page/[styleId]/sitemap.xml/route.ts` handler, and define a GET function there, which returns your sitemap: https://nextjs.org/docs/app/building-your-application/routing/route-handlers#non-ui-responses and using https://nextjs.org/docs/app/building-your-application/routing/route-handlers#dynamic-route-segments

And wait until the Next.js team adds support getting the dynamic paths on the special files. They might have reasons not to do that just yet. 🤷

**Reply by muezz** [reply]
The `route.ts` approach is interesting. That would give me access to the param even. I wonder why they chose to not give access to the params directly. I'll try this and see how it goes.

Thanks for the help.

**Reply by shawnmclean** [reply]
The route.ts path becomes unfeasible when we pair it with generateStaticParams. The vercel build throws:

Error: Invariant: failed to find source route /w/[username]/sitemap.xml for prerender /w/[username]/sitemap.xml at ignore-listed frames

I'm doing mulitenancy and don't want to spend compute on dynamic sitemap even if cached.
